### PR TITLE
[MooreToCore] Fix variable op lowering of aggregate types

### PR DESCRIPTION
**Pull request by maerhart**

### CODE REVIEW

File: lib/Conversion/MooreToCore/MooreToCore.cpp [hunk 1]
```diff
@@ -158,17 +158,26 @@ struct VariableOpConversion : public OpConversionPattern<VariableOp> {
   LogicalResult
   matchAndRewrite(VariableOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
     Type resultType = typeConverter->convertType(op.getResult().getType());
     Value init = adaptor.getInitial();
     // TODO: Unsupport x/z, so the initial value is 0.
     if (!init && cast<RefType>(op.getResult().getType()).getDomain() ==
                      Domain::FourValued)
       return failure();
```
Comment (fabianschuiki) [inline lib/Conversion/MooreToCore/MooreToCore.cpp:167]: I'm wondering whether we should remove this check and generally lower everything to an `i1` for now by mapping X and Z bits to 0. Verilator does a similar thing. Later, once we have proper support for four-valued types in HW/Comb/Seq, we could come back and adjust the lowering again.

**Review by fabianschuiki — APPROVED** [review]
Very nice!

File: lib/Conversion/MooreToCore/MooreToCore.cpp (hunk 1 above)
Comment (cepheus69) [reply lib/Conversion/MooreToCore/MooreToCore.cpp:167]: I support to choose X / Z pessimistic strategy. Once we support multi-valued types in core IR. A new option could be added to enable the optimistic strategy - spread x / z value and preserve their value.

Comment (maerhart) [reply lib/Conversion/MooreToCore/MooreToCore.cpp:167]: Adding a pass option to let the user decide whether x/z should be converted to 0 sounds like a reasonable approach to me. I'll leave it for a follow-up PR.

**Pull request merged**